File: Razmik/Week6/vigenere.py
#!/usr/bin/python3.7
from sys import argv, exit
if len(argv) != 2:
    exit("Enter valid")
if not argv[1].isalpha():
    exit("Enter valid")
key = [ord(i) % 32 - 1 for i in argv[1]]
a = input("plaintext:  ")
j = 0
print("ciphertext: ", end='')
for char in a:
    if char.isalpha():
        char = chr(ord(char) // 32 * 32 + (ord(char) % 32 + key[j % len(key)]) % 26)
        j += 1
    print(char, end='')
print()

File: Vachagan/CS50x 2019/Week 1 continued/vigenere/vigenere.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>

int main(int argc, string argv[])
{

    if (argc != 2)
    {
        printf("Please enter valid [KEY], ex. ./vigenere abc \n");
        return 1;
    }
    
    string keyword = argv[1];
    int keylength = strlen(keyword);
    for (int i = 0; i < keylength; i++) 
    {
        if (!isalpha(keyword[i]))
        {
            printf("Please enter valid [KEY], KEY only can alpha \n");
            return 1;
        }
    }

    string sentence = get_string("plaintext: ");

    int key = 0;
    int len = strlen(sentence);
    
    printf("ciphertext: ");
    for (int i = 0; i < len; i++) 
    {
        if (isalpha(sentence[i]))
        {
            int ascii = (sentence[i] < 97) ? 65 : 97;
            sentence[i] = (char)((sentence[i] - ascii + tolower(keyword[key++]) - 'a') % 26 + ascii);
        }
        printf("%c", sentence[i]);
        if (key == keylength)
        {
            key = 0;
        }
    }
    printf("\n");  
}